pkg/runner/ftp.go

```diff
@@ -401,15 +369,85 @@ func (fc *FtpClient) cpDir(src, dst string) (CommandResult, error) {
 }
 
 func (fc *FtpClient) cpFile(src, dst string) (CommandResult, error) {
-	cmd := exec.Command("cp", src, dst)
-	cmd.SysProcAttr = fc.sysProcAttr
-	if output, err := cmd.CombinedOutput(); err != nil {
+	err := copyFile(src, dst)
+	if err != nil {
 		return CommandResult{
-			Message: strings.ToLower(string(output)),
+			Message: err.Error(),
 		}, err
 	}
 
 	return CommandResult{
 		Message: fmt.Sprintf("Copy %s to %s", src, dst),
 	}, nil
 }
+
+func copyFile(src, dst string) error {
+	srcFile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer srcFile.Close()
```

Comment:
Ensure errors are handled explicitly when using defer (Go convention). Review and fix other areas as needed